### PR TITLE
8342270: Test sun/security/pkcs11/Provider/RequiredMechCheck.java needs write access to src tree

### PR DESCRIPTION
**Pull request by fguallini**
The test sun/security/pkcs11/Provider/RequiredMechCheck.java extends sun/security/pkcs11/PKCS11Test.java, which sets up a PKCS11Provider configured with NSS config files located in the source tree directory (sun/security/pkcs11/nss) for the dependent tests to run properly. This test currently fails if that **pkcs11/nss** directory is made read-only because these config files are updated during the test execution. Files located in the source tree should not be modified when running tests. 
This PR updates the PKCS11Test to copy the NSS config files to the Jtreg scratch directory and configures the provider to use those files instead.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342270](https://bugs.openjdk.org/browse/JDK-8342270): Test sun/security/pkcs11/Provider/RequiredMechCheck.java needs write access to src tree (**Bug** - P3)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) Review applies to [121de0aa](https://git.openjdk.org/jdk/pull/21665/files/121de0aa5186bf2a4c5d4eb8731e3ffa846496b5)
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21665/head:pull/21665` \
`$ git checkout pull/21665`

Update a local copy of the PR: \
`$ git checkout pull/21665` \
`$ git pull https://git.openjdk.org/jdk.git pull/21665/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21665`

View PR using the GUI difftool: \
`$ git pr show -t 21665`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21665.diff">https://git.openjdk.org/jdk/pull/21665.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21665#issuecomment-2432853308)
</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]

File: test/jdk/sun/security/pkcs11/PKCS11Test.java [hunk 1]
```diff
@@ -247,6 +247,30 @@ public static String getBase() throws Exception {
         return PKCS11_BASE;
     }
 
+    private static void copyFiles(Path source, String destination)
```
Comment (rhalade) [inline test/jdk/sun/security/pkcs11/PKCS11Test.java:250]: Check jdk.test.lib if there is a method you can use to copy files. If not then this method should be added to library.

File: test/jdk/sun/security/pkcs11/PKCS11Test.java [hunk 2]
```diff
@@ -497,13 +521,17 @@ public static String getNssConfig() throws Exception {
         }
 
         String base = getBase();
+        String nss = "nss";
+        Path nssDirSource = Path.of(base).resolve(nss);
+        String nssDirDestination = Path.of(".").resolve(nss).toString();
```
Comment (rhalade) [inline test/jdk/sun/security/pkcs11/PKCS11Test.java:526]: Using current directory here may be incorrect. Can this be changed to `Path.of(System.getProperty("test.classes"), ".")`

**Review by rhalade — CHANGES_REQUESTED** [review]

File: test/jdk/sun/security/pkcs11/PKCS11Test.java (hunk 2 above)
Comment (fguallini) [reply test/jdk/sun/security/pkcs11/PKCS11Test.java:526]: The "test.classes" directory contains the compiled classes according to [jtreg doc](https://openjdk.org/jtreg/tag-spec.html#testvars).
Whereas, the current directory is the [Jtreg scratch](https://openjdk.org/jtreg/faq.html#scratch-directory) directory, which is the right place for the tests temporary files. Also, Jtreg will automatically clean this directory when the test begins.
The scratch directory does seem like the best place to store these temporary configuration files.

File: test/jdk/sun/security/pkcs11/PKCS11Test.java (hunk 1 above)
Comment (fguallini) [reply test/jdk/sun/security/pkcs11/PKCS11Test.java]: Updated to use the existing Utils.copyFile, it was also needed to add the dependency in a shell test that uses PKCS11Test.java

**Pull request closed without merging**